syntax = "proto3";

package orders;

// Top-level service
service OrdersPricingService {
  // Returns the full order (line items, quantities, etc.)
  rpc GetOrder (GetOrderRequest) returns (GetOrderResponse);

  // Returns the total price for a given order
  rpc GetPrice (GetPriceRequest) returns (GetPriceResponse);
}

// --- Messages ---

message GetOrderRequest {
  string order_id = 1;
}

message LineItem {
  string sku   = 1;
  uint32 qty   = 2;
  double price = 3;
}

message GetOrderResponse {
  string order_id   = 1;
  repeated LineItem items = 2;
}

message GetPriceRequest {
  string order_id = 1;
}

message GetPriceResponse {
  string order_id = 1;
  double total_price = 2;
}
